### PR TITLE
docs: fix capitalization/style errors

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Fix a lot of capitalization inconsistencies.
- Fix spelling of Docker Composer -> Docker Compose.
- Add some monospaced text where I think it was missing.
- Use `Node.js` instead of `Node` in description field.

## Context:

I've tried to keep all changes organized into separate commits, so you can revert things you don't like easily.
The changes touch `docs/usage` and `docs/development`.

I hope I've done this right! :smile:

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

File: docs/usage/self-hosting.md [hunk 1]
```diff
@@ -254,13 +254,13 @@ Note: the GitHub.com token in env is necessary in order to retrieve Release Note
 
 You should save and test out this script manually first, and add it to cron once you've verified it.
 
-## Kubernetes for Gitlab, using Git over SSH
+## Kubernetes for GitLab, using Git over SSH
```
Comment (HonkingGoose) [inline docs/usage/self-hosting.md:257]: Question, does changing capitalization here break any link back to this section?

Comment (viceice) [reply docs/usage/self-hosting.md:257]: I don't think so, but now 100 % sure

File: docs/usage/docker.md [hunk 2]
```diff
@@ -146,7 +146,7 @@ module.exports = {
 };
 ```
 
-It is possible to add additional host rules following the [documentation](https://docs.renovatebot.com/configuration-options/#hostrules). For example if you have some images you host yourself that are password protected and also some images you pull from docker hub without authentication then you can configure for a specific Docker host like this:
+It is possible to add additional host rules following the [documentation](https://docs.renovatebot.com/configuration-options/#hostrules). For example if you have some images you host yourself that are password protected and also some images you pull from Docker hub without authentication then you can configure for a specific Docker host like this:
```
Comment (rarkins) [inline docs/usage/docker.md:149]:
```suggestion
It is possible to add additional host rules following the [documentation](https://docs.renovatebot.com/configuration-options/#hostrules). For example if you have some images you host yourself that are password protected and also some images you pull from Docker Hub without authentication then you can configure for a specific Docker host like this:
```

File: docs/usage/self-hosting.md (hunk 1 above)
Comment (rarkins) [reply docs/usage/self-hosting.md:257]: I think it's ok

**Review by rarkins — CHANGES_REQUESTED** [review]
Just one thing

File: docs/usage/docker.md (hunk 2 above)
Comment (HonkingGoose) [reply docs/usage/docker.md:149]: Good catch! :smile:

**Comment by HonkingGoose** [comment]
@rarkins I can't see your feedback on `docs/usage/self-hosting.md`, maybe because I committed your first suggestion for Docker Hub, and that then "resolves" the review...

**Comment by rarkins** [comment]
IIRC I only had one suggestion in the whole thing

**Comment by HonkingGoose** [comment]
Weird.. Maybe GitHub is acting up in some way. :confused: 
Just wanted to make sure I'm not missing something important. :smile:

**Review by viceice — APPROVED** [review]

**Comment by rarkins** [comment]
@HonkingGoose could you create a new PR to add yourself to this `package.json`'s contributors list (if you wish? I was going to add you but don't know which name or email to use.

**Pull request merged**